### PR TITLE
components & more

### PR DESCRIPTION
**Pull request by nionis**
re #15

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -1,16 +1,85 @@
...
+  }, []);
+
+  const handleSend = useCallback(async (content) => {
+    const sendMessage = async () => Promise.resolve();
+
+    sendMessage().then(() => {
+      setState(
+        produce((draft) => {
+          if (!draft.selectedConversation) return;
+          const conv = draft.conversations.get(draft.selectedConversation);
```
Comment (jjperezaguinaga) [inline pages/index.tsx:38]: Not sure whether `conv` is descriptive enough.

**Comment by jjperezaguinaga** [comment]
@nionis I see multiple times the following pattern `value ? setValue : undefined` (e.g. `right: isSmall ? undefined : "small"`), whereas I think `value && setValue` is a bit shorter and I believe equally descriptive. Any reason to keep the previous pattern as a preference moving forward?

File: src/state.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+// update state to a mocked state
+export const updateToMockState = () => {
+  const SELF = "16Uiu2HAm5ayHUYnv1tAQCrzceJgNnYzvWCZYiRthk4rF1rbY-self";
+  const COUNTERPARTY_A =
+    "16Uiu2HAm5ayHUYnv1tAQCrzceJgNnYzvWCZYiRt-counterpartyA";
+  const COUNTERPARTY_B =
+    "16Uiu2HAm5ayHUYnv1tAQCrzceJgNnYzvWCZYiRt-counterpartyB";
+
+  // cloning object as it breaks react? jesus..
+  const conversations: Map<string, Conversation> = new Map([
```
Comment (jjperezaguinaga) [inline src/state.ts:36]: React or TypeScript?

**Comment by nionis** [comment]
@jjperezaguinaga It's my personal preference, I dislike using `&&` as it results in `false` and not `undefined` which is what I want it to be, sometimes `false` is treated by `grommet` as disabling a prop, where with `undefined` it will use the default behavior

**Review by jjperezaguinaga — APPROVED** [review]
LGTM

File: pages/index.tsx (hunk 1 above)
Comment (nionis) [reply pages/index.tsx:38]: will update in another PR

File: src/state.ts (hunk 2 above)
Comment (nionis) [reply src/state.ts:36]: I think the issue is within immer TBH, will eventually replace these mocks with a better system

**Pull request merged**